includes/Core/DI/DI_Services_Aware_Trait.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\DI\DI_Services_Aware_Trait
+ *
+ * @package   Google\Site_Kit\Core\DI
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\DI;
+
+/**
+ * Trait for DI services aware classes.
+ *
+ * @since n.e.x.t
+ */
+trait DI_Services_Aware_Trait {
+
+	/**
+	 * Gets DI service instance.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $name Service name.
+	 * @return mixed DI service on success, otherwise NULL.
+	 */
+	public function __get( $name ) {
+		if ( $this instanceof DI_Aware_Interface ) {
+			return $this->get_di()->get( $name );
```

Comment:
Would be great to somehow ensure here that this only returns the value if it is actually a service. The trait includes "services-aware" as a term, so it probably shouldn't allow to just access _anything_ in the DI container from the outside?

Reply:
Ok, "services-aware" is probably incorrect naming for this trait. I have updated it to be `DI_Entry_Aware_Trait` to better reflect its purpose.